### PR TITLE
Added feature to start optimisation from a provided population

### PR DESCRIPTION
**Pull request by DrTaDa**
Proposition to solve issue #252

### CODE REVIEW

File: bluepyopt/deapext/optimisations.py [hunk 1]
```diff
@@ -258,7 +258,8 @@ def run(self,
             offspring_size=None,
             continue_cp=False,
             cp_filename=None,
-            cp_frequency=1):
+            cp_frequency=1,
+            population=[]):
```
Comment (wvangeit) [inline bluepyopt/deapext/optimisations.py:262]: Better to put None?

Comment (DrTaDa) [reply bluepyopt/deapext/optimisations.py:262]: Ok.

File: bluepyopt/deapext/optimisations.py [hunk 2]
```diff
@@ -268,7 +269,32 @@ def run(self,
             offspring_size = self.offspring_size
 
         # Generate the population object
-        pop = self.toolbox.population(n=offspring_size)
+        if population is not None:
+
+            if len(population) != offspring_size:
```
Comment (wvangeit) [inline bluepyopt/deapext/optimisations.py:274]: I think we might have to discuss here what we mean with 'population'. I'd say population = parents + offspring. Don't know how far that makes sense here. Should we maybe consider renaming the field people can specify 'offspring' ?

Comment (DrTaDa) [reply bluepyopt/deapext/optimisations.py]: Ok !

**Comment by wvangeit** [comment]
@anirban6908 This is a proposal by @DrTaDa to move the population restart feature to the run() call. Can you confirm that this code is what you expect the feature to do? Thanks.

**Comment by anirban6908** [comment]
@wvangeit Sorry for the late reply. Looks good if you want to start the optimization with a new population. 
For the PR #288 I was trying to ensure that if at the end of a checkpointed run (say Stage 1) you want to restart the optimization with a modified fitness function (say Stage 2), the statistics for this run will be written in the same logbook as in Stage 1 and the generation will count up from latest generation in Stage 1 - making it a single optimization run combined of both Stage 1 and Stage 2. 

Correct me if I am wrong but in this PR, for the situation I described the situation above, Stage 1 and Stage 2 will be treated as independent optimizations, will write in separate .pkl files (or even overwrite if checkpoint filename is not changed in Stage 2) - which I wanted to avoid since in my opinion this is the continuation of the same optimization (again for the situation I described).

**Comment by DrTaDa** [comment]
Hello @anirban6908,

**Comment by wvangeit** [comment]
@DrTaDa could you have a look at the merge conflict? thx

**Comment by DrTaDa** [comment]
@wvangeit The conflict is solved. Should we wait for the feedback from @anirban6908 before merging ?

**Comment by wvangeit** [comment]
I'll merge it now, but we can discuss further

**Comment by anirban6908** [comment]
Hi @DrTaDa Apologies for the delay. This looks good to me.

Thanks

**Comment by wvangeit** [comment]
@anirban6908 since this is now addressed in https://github.com/BlueBrain/BluePyOpt/pull/321, are you ok with closing this PR?

**Comment by anirban6908** [comment]
Done.

**Pull request merged**